var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -281,3 +285,47 @@ def cmake_args(self):
         options.append('-DBUILD_SHARED_LIBS=%s' % ('+shared' in self.spec))
 
         return options
+
+    test_script_relative_path = "scripts/spack_test"
+
+    @run_after('install')
+    def setup_build_tests(self):
+        # Skip if unsupported version
+        cmake_source_path = join_path(self.stage.source_path,
+                                      self.test_script_relative_path)
+        if not os.path.exists(cmake_source_path):
+            return
+        """Copy test."""
+        cmake_out_path = join_path(self.test_script_relative_path, 'out')
+        cmake_args = [cmake_source_path,
+                      "-DSPACK_PACKAGE_SOURCE_DIR:PATH={0}".
+                      format(self.stage.source_path),
+                      "-DSPACK_PACKAGE_TEST_ROOT_DIR:PATH={0}".
+                      format(join_path(self.install_test_root, cmake_out_path)),
+                      "-DSPACK_PACKAGE_INSTALL_DIR:PATH={0}".format(self.prefix)]
+        cmake(*cmake_args)
+        self.cache_extra_test_sources(cmake_out_path)
+
+    def build_tests(self):
+        """Build test."""
+        cmake_path = join_path(self.install_test_root,
+                               self.test_script_relative_path, 'out')
+        cmake_args = [cmake_path, '-DEXECUTABLE_OUTPUT_PATH=' + cmake_path]
+        cmake(*cmake_args)
+        make()
+
+    def run_tests(self):
+        """Run test."""
+        reason = 'Checking ability to execute.'
+        run_path = join_path(self.install_test_root,
+                             self.test_script_relative_path, 'out')
+        self.run_test('make', [run_path, 'test'], [], installed=False, purpose=reason)
+
+    def test(self):
+        # Skip if unsupported version
+        cmake_path = join_path(self.install_test_root,
+                               self.test_script_relative_path, 'out')
+        if not os.path.exists(cmake_path):
```

Comment:
```suggestion
        if not os.path.exists(cmake_path):
            print('Skipping smoke tests: {0} is missing'.format(cmake_path))
```

Reply:
As written, the test log has no information on the smoke test when the script is missing other than indicating the test passed, giving the impression everything went fine even thought the smoke test was skipped.  For example, I'm getting the following:

```
==> Results for test suite 'kokkos':
==>   kokkos-3.3.01-xjr6jkq PASSED
==> Testing package kokkos-3.3.01-xjr6jkq
```

Even when the smoke test is skipped.

While this may be desirable in terms of not claiming failure if the smoke test cannot run, it can be misleading; hence, the suggestion of noting when the smoke test is skipped.